clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientBuilder.java

```diff
@@ -60,6 +61,34 @@ ZeebeClientBuilder applyEnvironmentVariableOverrides(
    */
   ZeebeClientBuilder numJobWorkerExecutionThreads(int numThreads);
 
+  /**
+   * Identical behavior as {@link #jobWorkerExecutor(ScheduledExecutorService,boolean)}, but taking
+   * ownership of the executor by default. This means the given executor is closed when the client
+   * is closed.
+   *
+   * @param executor an executor service to use when invoking job workers
+   * @see #jobWorkerExecutor(ScheduledExecutorService, boolean)
+   */
+  default ZeebeClientBuilder jobWorkerExecutor(final ScheduledExecutorService executor) {
+    return jobWorkerExecutor(executor, true);
+  }
+
+  /**
+   * Allows passing a custom executor service that will be shared by all job workers created via
+   * this client.
+   *
+   * <p>Polling and handling jobs (e.g. via {@link io.camunda.zeebe.client.api.worker.JobHandler}
+   * will all be invoked on this executor.
+   *
+   * <p>When non-null, this setting override {@link #numJobWorkerExecutionThreads(int)}.
```

Comment:
🤔 WDYT about adding a check to prevent users from using both? Throw an `IllegalArgumentException` when they try to change both the values to prevent mistakes.

Reply:
I would be worried that it may affect deployments of clients using properties, e.g. use these default properties, but here's a handful of overrides for this specific deployment. For example, with the Spring Boot client.

I don't really have any usage data to support this though, so I would defer to whatever UX your team prefers :upside_down_face: